lib/ansible/modules/packaging/os/rhsm_repository.py

```diff
@@ -0,0 +1,263 @@
+        list or a YAML list.
+      - If state is equal to enabled or disabled, this option is required.
+      - If state is not equal to enabled or disabled, this option must not be used.
+    required: False
+    default: None
+'''
+
+EXAMPLES = '''
+- name: List all RHSM repositories.
+  rhsm_repository:
+    state: list
+  register: rhsm_repository
+
+- name: List enabled RHSM repositories.
+  rhsm_repository:
+    state: list_enabled
+  register: enabled_rhsm_repository
+
+- name: Enable a RHSM repository
+  rhsm_repository:
+    name: rhel-7-server-rpms
+
+- name: Disable all RHSM repositories
+  rhsm_repository:
+    name: '*'
+    state: disabled
+
+- name: Enable all repositories starting with rhel-6-server
+  rhsm_repository:
+    name: rhel-6-server*
+    state: enabled
+
+- name: Disable all repositories except rhel-7-server-rpms
+  rhsm_repository:
+    name: "{{ item }}"
+    state: disabled
+  with_items: "{{
+    rhsm_repository.repositories |
+    map(attribute='id') |
+    difference(['rhel-7-server-rpms']) }}"
+'''
+
+RETURN = '''
+repositories:
+  description:
+    - The list of RHSM repositories with their states.
+    - When this module is used to change the repositories states, this list contains the updated states after the changes.
+  returned: success
+  type: list
+'''
+
+import re
+import os
+from fnmatch import fnmatch
+from copy import deepcopy
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_subscription_manager(module, arguments):
+    # Execute subuscription-manager with arguments and manage common errors
+    rhsm_bin = module.get_bin_path('subscription-manager')
+    if not rhsm_bin:
+        module.fail_json(msg='The executable file subscription-manager was not found in PATH')
+
+    rc, out, err = module.run_command("%s %s" % (rhsm_bin, " ".join(arguments)))
+
+    if rc == 1 and (err == 'The password you typed is invalid.\nPlease try again.\n' or os.getuid() != 0):
+        module.fail_json(msg='The executable file subscription-manager must be run using root privileges')
+    elif rc == 0 and out == 'This system has no repositories available through subscriptions.\n':
+        module.fail_json(msg='This system has no repositories available through subscriptions')
+    elif rc == 1:
+        module.fail_json(msg='subscription-manager failed with the following error: %s' % err)
+    else:
+        return rc, out, err
+
+
+def get_repository_list(module, list_parameter):
+    # Generate RHSM repository list and return a list of dict
+    if list_parameter == 'list_enabled':
+        rhsm_arguments = ['repos', '--list-enabled']
+    elif list_parameter == 'list_disabled':
+        rhsm_arguments = ['repos', '--list-disabled']
+    elif list_parameter == 'list':
+        rhsm_arguments = ['repos', '--list']
+    rc, out, err = run_subscription_manager(module, rhsm_arguments)
+
+    skip_lines = ['+----------------------------------------------------------+',
```

Comment:
Please format this like this:

```python
    skip_lines = [
        '+----------------------------------------------------------+',
        '    Available Repositories in /etc/yum.repos.d/redhat.repo'
    ]
```

Reply:
Changed in the last commit, thanks for the review